base/ltmiscen.dtx

```diff
@@ -1306,11 +1325,23 @@
 %
 %  \begin{environment}{verbatim}
 %  \begin{macro}{\verbatim}
+% \changes{v1.2b}{2023/06/14}{Support tabs}
 %  \begin{macro}{\endverbatim}
 %    (RmS 93/09/19) Protected against `missing item' error message
 %               triggered by empty verbatim environment.
 %    \begin{macrocode}
-\def\verbatim{\@verbatim \frenchspacing\@vobeyspaces \@xverbatim}
+%</2ekernel>
+%<latexrelease>\IncludeInRelease{2023/11/01}%
+%<latexrelease>                 {\verbatim}{Obeyed tabs}%
+%<*2ekernel|latexrelease>
+\def\verbatim{\@verbatim \frenchspacing\@vobeyspaces\@vobeytabs \@xverbatim}
+%</2ekernel|latexrelease>
+%<latexrelease>\EndIncludeInRelease
+%<latexrelease>\IncludeInRelease{0000/00/00}%
+%<latexrelease>                 {\verbatim}{Obeyed tabs}%
+%<latexrelease>\def\verbatim{\@verbatim \frenchspacing\@vobeyspaces \@xverbatim}
+%<latexrelease>\EndIncludeInRelease
+%<*2ekernel>
```

Comment:
if tabs are handled inside \@vobeyspaces then this hsouldn't be needed, should it?

Reply:
I'd wondered about that. However, that leaves you vulnerable -  to a re-definition of `\@vobeyspaces`, as that would also kill tab support. That *might* of course be the position we want to take. So I've backed out my change and will look again at this - there's also something not quite right and I want to understand that too.

Reply:
the way I see it is this: we interpret `\@vobeyspaces` as "handle all space like chars (space TAB, maybe others one day) for verbatim" and  `\@vobeytabs` as an indirection to do something special to tabs alone on top of it. To my knowledge there is no redefinition of `\vobeyspaces` right now and if there is one in the future it has to take care of the assumption.

Reply:
See if you like the new version